### PR TITLE
[sc-543] Add hvml test on example programs

### PR DESCRIPTION
**Pull request by imaqtkatt**

### CODE REVIEW

**Review by developedby — COMMENTED** [review]
Maybe instead we could have these files in the examples directory and the test function reads from there?
This way we test the examples and are forced to keep them up-to-date, give users more examples and don't duplicate these programs.

**Comment by developedby** [comment]
That's what hvm-core is doing I think, if you want to have a look there.

File: tests/golden_tests/examples/all_tree.hvm
Comment (developedby) [inline tests/golden_tests/examples/all_tree.hvm:1]: The files in this directory are not being used, right?

Comment (imaqtkatt) [reply tests/golden_tests/examples/all_tree.hvm:1]: Yes, I thought I moved then but it was copied instead.

**Review by developedby — APPROVED** [review]

**Pull request merged**